src/snyk/common/proxy.ts

```diff
@@ -77,3 +93,37 @@ export function getProxyEnvVariable(
   // noinspection HttpUrlsUsage
   return `${protocol}//${auth ? `${auth}@` : ''}${host}${port ? `:${port}` : ''}`;
 }
+
+function getHttpsAgent(configuration: IConfiguration, logger: ILog): Agent {
+  return new Agent({
+    ...getDefaultAgentOptions(configuration, logger),
+  });
+}
+
+function getDefaultAgentOptions(
+  configuration: IConfiguration,
+  logger: ILog,
+  processEnv: NodeJS.ProcessEnv = process.env,
+): AgentOptions | undefined {
+  let defaultOptions: AgentOptions | undefined;
+
+  const sslCheck = !configuration.getInsecure();
+  defaultOptions = { rejectUnauthorized: sslCheck };
+  if (!sslCheck) {
+    globalAgent.options.rejectUnauthorized = false;
+  } else {
+    // use custom certs if provided
+    if (processEnv.NODE_EXTRA_CA_CERTS) {
+      try {
+        fs.accessSync(processEnv.NODE_EXTRA_CA_CERTS);
```

Comment:
It's generally better to use async pattern when working with file system to prevent from blocking the main thread. There should be `fs/promises` module you can pull the async function from.